docs/src/main/sphinx/security/cert-auth.rst

```diff
@@ -0,0 +1,108 @@
+==========================
+Certificate authentication
+==========================
+
+Trino can be configured to support certified clients seeking validation from a
+Trino server. This authentication method is only provided to support sites that
+have an absolute requirement for client authentication *and already have* client
+certificates for each client. Sites in this category have an existing PKI
+infrastructure, possibly including an onsite CA.
+
+This feature is *not* appropriate for sites that would need to generate a set of
+client certificates in order to use this authentication method. Starburst
+recommends instead configuring one of the password-based authentication methods,
+including :doc:`password-file`, :doc:`ldap`, and :doc:`salesforce`; or choosing
+an :doc:`OAuth2 <oauth2>`-compatible authentication vendor.
+
+
+Using certificate authentication
+--------------------------------
+
+With certificate authentication enabled, the validation sequence is the
+following:
+
+* A client verfies the coordinator's certificate, and makes an attempt to
+  connect to the coordinator.
+* The coordinator asks the client for its certificate.
+* The client responds with *its* certificate, validated by *its* trust store.
+* The coordinator verifies the certificate against that trust store.
+
+The server’s trust store must include the certificate for the signer of the
+client certificates, and the client’s trust store must include the certificate
+for the signer of the server’s certificate. They do not need to be the same
+signer certificate.
+
+Trino validates based on the distinguished name (DN) from the X.509 Subject
+field. The principal created for the connection is the common name (CN) from the
+Subject field. You can use :doc:`User mapping </security/user-mapping>` to map
+the CN to a Trino user name.
+
+There are three levels of client certificate support possible. From the point of
+view of the server:
+
+* I do not need a certificate from clients
+* I would like to have a certificate
+* I must have a certificate
+
+Trino's client certificate support is the middle one, "would like to have."
+
+Certificate authentication configuration
```

Comment:
Configuration

Reply:
not sure we want to duplicate the page title.. typically that is not done in other docs

Reply:
OK